packages/mermaid/src/accessibility.spec.ts

```diff
@@ -0,0 +1,218 @@
+import { MockedD3 } from './tests/MockedD3';
+import { setA11yDiagramInfo, addSVGa11yTitleDescription } from './accessibility';
+
+describe('accessibility', () => {
+  const fauxSvgNode = new MockedD3();
+
+  describe('setA11yDiagramInfo', () => {
+    it('sets the aria-roledescription to the diagram type', () => {
+      // @ts-ignore  Required to easily handle the d3 select types
+      const svg_attr_spy = vi.spyOn(fauxSvgNode, 'attr').mockReturnValue(fauxSvgNode);
+      setA11yDiagramInfo(fauxSvgNode, 'flowchart');
+      expect(svg_attr_spy).toHaveBeenCalledWith('aria-roledescription', 'flowchart');
+    });
+
+    it('does nothing if the diagram type is empty', () => {
+      // @ts-ignore  Required to easily handle the d3 select types
+      const svg_attr_spy = vi.spyOn(fauxSvgNode, 'attr').mockReturnValue(fauxSvgNode);
+      setA11yDiagramInfo(fauxSvgNode, '');
+      expect(svg_attr_spy).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('addSVGa11yTitleDescription', () => {
+    const givenId = 'theBaseId';
+
+    describe('with the given svg d3 object:', () => {
+      it('does nothing if there is no insert defined', () => {
+        const noInsertSvg = {
+          attr: vi.fn(),
+        };
+        const noInsert_attr_spy = vi.spyOn(noInsertSvg, 'attr').mockReturnValue(noInsertSvg);
+        addSVGa11yTitleDescription(noInsertSvg, 'some title', 'some desc', givenId);
+        expect(noInsert_attr_spy).not.toHaveBeenCalled();
+      });
+
+      // ----------------
+      // Convenience functions to DRY up the spec
+
+      function expectAriaLabelledByIsTitleId(
+        svgD3Node: any,
```

Comment:
Is it worth using the following type instead of `any` (copied from `packages/mermaid/src/accessibility.ts`):

```ts
// This is just a convenience alias to make it clear the type is a d3 object. (It's easier to make it 'any' instead of the complete typing set in d3)
type D3object = any;
```

It will make it easier to update the types later on.

I think something like the following _might_ work too, but since I'm not 100% sure, it might just be easier to leave it as `any`:

```typescript
import {type Selection} from 'd3';
type D3SvgObject = Selection<SVGSVGElement, any, any, any>;
```

Reply:
Agree.  Exported the type from mermaidAPI (the 1 place we can change it later) for both accessibility.ts and the .spec